google/cloud/bigtable/data/_helpers.py

```diff
@@ -62,6 +63,51 @@ def _attempt_timeout_generator(
         yield max(0, min(per_request_timeout, deadline - time.monotonic()))
 
 
+def _exponential_sleep_generator(
```

Comment:
This is a [generator function](https://www.geeksforgeeks.org/generators-in-python/), so there is no risk of an infinite loop. The method "returns" at the yield line, only to pick up there again the next time `next(this_generator)` is called.

If the min increase is 0 and the multiplier is 1, it will continuously spit out the `initial` value every time you call `next`, but won't block